src/Nethermind/Nethermind.Consensus.AuRa/AuRaPlugin.cs

```diff
@@ -77,9 +77,35 @@ public Task<IBlockProducer> InitBlockProducer(IBlockProductionTrigger? blockProd
 
         public IBlockProductionTrigger? DefaultBlockProductionTrigger { get; private set; }
 
-        public INethermindApi CreateApi(IConfigProvider configProvider, IJsonSerializer jsonSerializer,
-            ILogManager logManager, ChainSpec chainSpec) => new AuRaNethermindApi(configProvider, jsonSerializer, logManager, chainSpec);
+        public IModule? GetModule(string engineType, IConfigProvider configProvider)
+        {
+            if (engineType != SealEngineType) return null;
+            return new AuraModule();
+        }
 
-        public bool ShouldRunSteps(INethermindApi api) => true;
+        public class AuraModule : Module
+        {
+            protected override void Load(ContainerBuilder builder)
+            {
+                base.Load(builder);
+
+                builder.RegisterType<AuRaNethermindApi>()
+                    .AsSelf()
+                    .As<INethermindApi>()
+                    .SingleInstance();
+
+                builder.RegisterDecorator<IGasLimitCalculator>((ctx, _, baseGasLimit) =>
+                {
+                    // So aura does a strange thing where the gas limit calculator is replaced later on. Not sure exactly
+                    // why gas limit calculator is normally declared very early on. In any case, since its gas limit
+                    // calculator is very complicated, it can't be resolved until more of the stack is migrated to DI.
+                    AuRaNethermindApi api = ctx.Resolve<AuRaNethermindApi>();
+                    if (api.AuraGasLimitCalculator != null) return api.AuraGasLimitCalculator;
+                    return baseGasLimit;
```

Comment:
It is because gas calculation can be dependent on system transaction call, so you need to initiate transaction processor first ect.